src/prng.rs

```diff
@@ -60,15 +61,14 @@ fn random_field_from_seed<F: FieldElement>(seed: &[u8], length: usize) -> Vec<F>
         cipher.apply_keystream(&mut buffer);
 
         for chunk in buffer.chunks_exact(F::BYTES) {
-            match F::read_from(chunk) {
+            match bincode::deserialize(chunk) {
                 Ok(x) => {
                     output[output_written] = x;
                     output_written += 1;
                     if output_written == length {
                         break;
                     }
                 }
-                Err(FieldError::FromBytesModulusOverflow) => (), // reject this sample
```

Comment:
Handling errors is kind of ugly. Since bincode [implements it's own ErrorKind](https://docs.rs/bincode/1.3.3/bincode/enum.ErrorKind.html), errors propagate from serde into the `Custom(String)` kind after formatting the error via the `Display` trait. This means that pattern matching on the specific deserialization error can't be done through the guard.